samples/gui/lvgl/src/main.c

```diff
@@ -37,7 +43,7 @@ void main(void)
 	lv_obj_t *hello_world_label;
 	lv_obj_t *count_label;
 
-	display_dev = device_get_binding("ILI9340");
```

Comment:
IMHO display API should be generic enough to make this line sole change in the sample.
(In this case changing SDL pixel format should be enough).

Reply:
It is not clear here what you mean exactly:
1) take care of the conversion in both, SDL and ILI9340, display drivers
-> In this case best put discuss this in display driver API pull request (#7833)
2) Extend the kconfig options of the SDL display driver to select pixel format at compile time
-> This could lead to a lot of options in the config
3) Detect pixel format at run time for the SDL display
-> Possibly the best solution, but for simplicity only detect format on first write

Reply:
Ideally the user of display API should not change behaviour depending on display type. For example, the same code should correctly drive B&W and color display as long as both uses the same resolution. Of course performance and image quality will be compromised in such case, as some pixel format conversion will be needed. However it "should work" in my opinion.

I do not know on what level of display API such conversion/abstraction might implemented. Basing on this PR I assume that display.h just abstracts hardware and the LittlevGL provides high level primitives, thus the code use LittlevGL should be able to drive any display (especially if there will be API to fetch basic display properties like resolution and the sample will use this API in order to create correct picture).

Reply:
Will be handled in PR #6826 by the GUI library